### DISCUSSION TITLE
How to go directly to provider signin?

### DISCUSSION THREAD

**Discussion by awildboop** [nextauthjs/next-auth#2174] [Help] [answered]
In my project, I am only using Auth0 for authentication. Instead of going to the signin page and clicking "Sign in with Auth0", how would I just skip straight to the Auth0 login screen? I've tried using /api/auth/signin/auth0, but all it does is show the auth0 button instead of going to the auth0 signin screen.

**Comment by balazsorban44** [comment, marked as answer]
https://next-auth.js.org/getting-started/client#starts-google-oauth-sign-in-flow-when-clicked

`signIn("auth0")`

**Reply by awildboop** [reply]
Perfect, that's exactly what I wanted! Thanks!

**Reply by subhendukundu** [reply]
I have a question about this.
```
<button
    onClick={() =>
      signIn("google", {
        redirect: true,
        callbackUrl: "http://localhost:3000/projects",
      })
    }
  >
    <Google />
  </button>
  ```
  
  When I call this, this redirects me to `http://localhost:4000/api/auth/signin?csrf=true`, which is showing the providers' buttons. Would it be possible to redirect directly to go to google auth page?

**Reply by sureshimp766** [reply]
the above does not work for Cognito. it still take me to the list of providers

**Reply by balazsorban44** [reply]
@subhendukundu and @sureshimp766 you will have to provide a reproduction. It is probably just a tiny misconfiguration somewhere.

**Reply by subhendukundu** [reply]
Aha, I thought I added the repo.
https://github.com/subhendukundu/hasura-nextjs-auth

Also there is one more thing bothering me for a whole. I have multiple app which could share the same auth system. Would that be possible? I mean, if I would like to access this from another subdomain? I did saw few topics, but couldn't understood much. Definitely need to change the cookie settings.
